source/gateway.go

```diff
@@ -117,7 +117,7 @@ func NewIstioGatewaySource(
 // Endpoints returns endpoint objects for each host-target combination that should be processed.
 // Retrieves all gateway resources in the source's namespace(s).
 func (sc *gatewaySource) Endpoints() ([]*endpoint.Endpoint, error) {
-	gwList, err := sc.istioClient.NetworkingV1alpha3().Gateways(sc.namespace).List(metav1.ListOptions{})
+	gwList, err := sc.istioClient.NetworkingV1alpha3().Gateways(sc.namespace).List(context.Background(), metav1.ListOptions{})
```

Comment:
what about adding the context in the gatewaySource?

Reply:
I can look at that sure.

Reply:
Thinking about this more I think there is a larger task to plumb down the context from main -> store -> sources. I think that makes more sense as a follow up to this but I could do it in this PR if that is desired.

Reply:
@njuettner this was done.